File: apis/backend/src/main/java/com/tms/api/objects/models/project/ProjectModel.java
package com.tms.api.objects.models.project;

import java.util.Date;
import java.util.List;

import com.tms.api.objects.models.user.UserPermissionModel;
import com.tms.api.objects.types.StatusType;

public record ProjectModel(
    String projectId,
    String organizationId,
    String projectGuid,
    String projectName,
    List<UserPermissionModel> Users,
    StatusType statusId,
    Date createdDate,
    Date modifiedDate
) {}

File: ticket-managment-system-web/src/app/core/models/project/project.model.ts
import { UserPermissionModel } from "..";
import { StatusType } from "../../enums";

export class ProjectModel {
    public projectId!: string;
    public organizationId!: string;
    public projectGuid!: string;
    public projectName!: string;
    public Users: UserPermissionModel[] = [];
    public SprintIds!: string;
    public statusId!: StatusType;

    constructor(project: Partial<ProjectModel> = {}) {
        Object.assign(this, project);
    }
}